applications/GeoMechanicsApplication/python_scripts/geomechanics_T_solver.py

```diff
@@ -0,0 +1,203 @@
+    # =============================================================================================
+    def __init__(self, model, custom_settings):
+        super().__init__(model, custom_settings)
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Construction of Solver finished.")
+
+    # =============================================================================================
+    # =============================================================================================
+    @classmethod
+    def GetDefaultParameters(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "solver_type": "geomechanics_T_solver",
+            "model_part_name": "PorousDomain",
+            "domain_size": 2,
+            "start_time": 0.0,
+            "model_import_settings":{
+                "input_type": "mdpa",
+                "input_filename": "unknown_name"
+            },
+            "material_import_settings" :{
+                "materials_filename": ""
+            },
+            "time_stepping": {
+                "end_time" : 1.0,
+                "time_step": 0.1
+            },
+            "buffer_size": 2,
+            "echo_level": 0,
+            "rebuild_level": 2,
+            "reform_dofs_at_each_step": false,
+            "clear_storage": false,
+            "compute_reactions": false,
+            "move_mesh_flag": false,
+            "nodal_smoothing": false,
+            "reset_displacements":  false,
+            "solution_type": "quasi_static",
+            "scheme_type": "Newmark",
+            "newmark_beta": 0.25,
+            "newmark_gamma": 0.5,
+            "newmark_theta": 0.5,
+            "strategy_type": "newton_raphson",
+            "convergence_criterion": "temperature_criterion",
+            "temperature_relative_tolerance": 1.0e-4,
+            "temperature_absolute_tolerance": 1.0e-9,
+            "residual_relative_tolerance": 1.0e-4,
+            "residual_absolute_tolerance": 1.0e-9,
+            "desired_iterations"         : 4,
+            "max_radius_factor"          : 20.0,
+            "min_radius_factor"          : 0.5,
+            "max_iterations"             : 15,
+            "min_iterations"             : 6,
+            "number_cycles"              : 5,
+            "increase_factor"            : 2.0,
+            "reduction_factor"           : 0.5,
+            "calculate_reactions"        : true,
+            "max_line_search_iterations" : 5,
+            "first_alpha_value"          : 0.5,
+            "second_alpha_value"         : 1.0,
+            "min_alpha"                  : 0.1,
+            "max_alpha"                  : 2.0,
+            "line_search_tolerance"      : 0.5,
+            "rotation_dofs"              : false,
+            "block_builder"              : true,
+            "search_neighbours_step"     : false,
+            "linear_solver_settings":{
+                "solver_type": "AMGCL",
```

Comment:
just for my curiosity, are you using by default AMGCL? does it performs well in your solid mechanics cases?

Reply:
for pressure and deformation-pressure, this solver is applied. In order to be consistent with the other parts of the code, I kept it as AMGCL.

However, AMGCL has some advatages. The multigrid scales linearly with the number of nodes. However, Krylove space methods scale exponentially. Then for large number of nodes, multigrid is more efficient. from other side, for small number of nodes Krylove is faster. As we assume, we usually have large number of nodes, per default AMGCL is "probably" here set as the default solver.